qemu/tests/block_hotplug_in_pause.py

```diff
@@ -0,0 +1,177 @@
+        elif params.get("os_type") == "windows":
+            pattern = r"^\d+"
+        else:
+            test.cancel("Unsupported OS type '%s'" % params.get("os_type"))
+
+        output = session.cmd_output_safe(cmd)
+        disks = re.findall(pattern, output, re.M)
+        return disks
+
+    def get_plug_unplug_disks(disk1, disk2):
+        """
+        Get the different disk between disk1 and disk2.
+        """
+        disk = list(set(disk2) ^ (set(disk1)))
+        return disk
+
+    def block_hotplug(vm, image_name):
+        """
+        Hotplug disks and verify it in qtree
+        """
+        image_params = params.object_params(image_name)
+        devs = vm.devices.images_define_by_params(image_name,
+                                                  image_params, 'disk')
+        for dev in devs:
+            ret = vm.devices.simple_hotplug(dev, vm.monitor)
+            if ret[1] is False:
+                test.fail("Failed to hotplug device '%s'."
+                          "Output:\n%s" % (dev, ret[0]))
+        devs = [dev for dev in devs if not isinstance(dev, qdevices.QDrive)]
+        return devs
+
+    def block_unplug(vm, device_list):
+        """
+        Unplug disks and verify it in qtree
+        """
+        for dev in reversed(device_list):
+            ret = vm.devices.simple_unplug(dev, vm.monitor)
+            if ret[1] is False:
+                test.fail("Failed to unplug device '%s'."
+                          "Ouptut:\n%s" % (dev, ret[0]))
+
+    def block_check_in_guest(session, disks, blk_num,
+                             get_disk_cmd, plug_tag="hotplug"):
+        """
+        Check hotplug/unplug disks in guest
+        """
+        logging.info("Check block device in guest after %s." % plug_tag)
+        pause = float(params.get("virtio_block_pause", 30.0))
+        status = utils_misc.wait_for(lambda: len(get_plug_unplug_disks(disks,
+                                     find_disk(session, get_disk_cmd))) == blk_num,
+                                     pause)
+        disks = get_plug_unplug_disks(disks, find_disk(session, get_disk_cmd))
+        if not status:
+            test.fail("Failed to %s device to guest, expected: %d,"
+                      "actual: %d" % (plug_tag, blk_num, len(disks)))
+
+    def rw_disk_in_guest(session, plug_disks, iteration):
+        """
+        Do read/write on hotplug disks
+        """
+        logging.info("Read/Writ on block device after hotplug.")
+        disk_op_timeout = int(params.get("disk_op_timeout", 360))
+        disk_letter = params.objects("disk_letter")
+        for disk in plug_disks:
+            if params.get("os_type") == "linux":
+                test_cmd = params.get("disk_op_cmd") % (disk, disk)
+            elif params.get("os_type") == "windows":
+                if iteration == 0:
+                    error_context.context("Format disk", logging.info)
+                    utils_misc.format_windows_disk(session, disk,
```

Comment:
here, may use  utils_disk.configure_empty_disk() replace format_windows_disk() is better, the func  utils_disk.configure_empty_disk() can return the mountpoint list, like ['E', 'F'], thus, will no need to define the disk_letter = "I J" in cfg file, it can be automatic assign letter for new disk, and also can simple 105 line %(disk_letter[int(disk)-1], disk_letter[int(disk)-1])). like:

disk_letter = utils_disk.configure_empty_disk()
test_cmd = params.get("disk_op_cmd") % (disk_letter[0], disk_letter[1])